### ISSUE TITLE
Resource management for simultaneous requests 

### ISSUE THREAD

**Issue by MortezaBashsiz** [MortezaBashsiz/nipovpn#65]
### Issue 
The program is not able to handle simultaneous requests.

### Produce the issue
To produce this issue, it is enough to run a while loop to send a request from two different terminals at the same time. Like following

```bash
[~/Temp]>$ curl -L https://speed.cloudflare.com/__down?bytes=10& curl -L https://www.google.com&
[1] 2856819
[2] 2856820
[~/Temp]>$ curl: (60) SSL: no alternative certificate subject name matches target hostname 'speed.cloudflare.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the webpage mentioned above.
curl: (35) OpenSSL/3.3.1: error:0A0000F4:SSL routines::unexpected message

[1]-  Exit 60                 curl -L https://speed.cloudflare.com/__down?bytes=10
[2]+  Exit 35                 curl -L https://www.google.com
[~/Temp]>$ 
```
Then you will see that both of them will be failed without any progress

**Comment by MortezaBashsiz** [comment]
At the moment I am trying to fix it in this **[branch](https://github.com/MortezaBashsiz/nipovpn/tree/morteza/threads)**

**Comment by MortezaBashsiz** [comment]
Fixed but PR #108 
Simply creating tcpClient per connection fixed the issue
